ui/app/helpers/create-columns.js

```diff
@@ -0,0 +1,7 @@
+import { helper } from '@ember/component/helper';
+
+function createColumns(positional) {
+  return positional.map((column) => ({ label: column }));
+}
+
+export default helper(createColumns);
```

Comment:
I'm not convinced we want a helper just to make tables work with the fields we want, but if we decided we did:
- I'd like this made more generalized (like `map-to-objects` instead of `create-columns`)
- a second argument (either positional or named) for `key`, which would let you customize what you currently have hardcoded as `label` currently

My first suggestion would still be to consider if we really want to go down the road of invoking a helper every time we display a table, though.

Reply:
Roger that. Normally we put this logic in our controllers check any controller that is using the `SingleSelectDropdown` feature and you'll see redundant code. IMHO the code isn't adding value when its in the `controller` or `component` because its not real state, just data transformation for working with another component API.

You'll see logic like [this](https://github.com/hashicorp/nomad/blob/main/ui/app/controllers/jobs/index.js#L84-L103) across 26 different views (about 50% of our views).

On pages where we have multiple dropdowns, it really muddies the controller and makes it challenging to find the code that matters. Most of the code in this [file](https://github.com/hashicorp/nomad/blob/main/ui/app/controllers/jobs/index.js) is just derived state that matches the component API for `SingleSelectDropdown` and all the code boils down to is iterating over a list to get a `key` and `label`.